packages/core/src/components/button/buttons.tsx

```diff
@@ -73,13 +87,27 @@ export const AnchorButtonFactory = React.createFactory(AnchorButton);
 function getButtonClasses(props: IButtonProps) {
     return classNames(
         Classes.BUTTON,
-        { [Classes.DISABLED]: props.disabled },
+        { [Classes.DISABLED]: isButtonDisabled(props) },
         Classes.iconClass(props.iconName),
         Classes.intentClass(props.intent),
         props.className,
     );
 }
 
+function isButtonDisabled(props: IButtonProps) {
+    return props.disabled === true || props.isLoading === true;
```

Comment:
To answer your question, yes, I'd also expect a "loading" button to be disabled. Let's go with that 👍